libs/s25main/figures/nofMiner.cpp

```diff
@@ -71,20 +76,100 @@ helpers::OptionalEnum<GoodType> nofMiner::ProduceWare()
 
 bool nofMiner::AreWaresAvailable() const
 {
-    return nofWorkman::AreWaresAvailable() && FindPointWithResource(GetRequiredResType()).isValid();
+    if(!nofWorkman::AreWaresAvailable())
+        return false;
+
+    const MiningBehavior addonSetting = GetMiningBehavior();
+
+    if(addonSetting == MiningBehavior::AlwaysAvailable)
+        return true;
+
+    // TODO: Skip this check if siv like mines and worked at least once?
+    if(addonSetting == MiningBehavior::S4Like)
+        return FindPointWithResource(GetRequiredResType(), MINER_RADIUS_SETTLERSIV).isValid();
+    else
+        return FindPointWithResource(GetRequiredResType()).isValid();
 }
 
-bool nofMiner::StartWorking()
+MiningBehavior nofMiner::GetMiningBehavior() const
 {
-    MapPoint resPt = FindPointWithResource(GetRequiredResType());
-    if(!resPt.isValid())
-        return false;
     const GlobalGameSettings& settings = world->GetGGS();
-    bool inexhaustibleRes = settings.isEnabled(AddonId::INEXHAUSTIBLE_MINES)
-                            || (workplace->GetBuildingType() == BuildingType::GraniteMine
-                                && settings.isEnabled(AddonId::INEXHAUSTIBLE_GRANITEMINES));
-    if(!inexhaustibleRes)
-        world->ReduceResource(resPt);
+
+    switch(workplace->GetBuildingType())
+    {
+        case BuildingType::GoldMine: return (MiningBehavior)settings.getSelection(AddonId::MINING_OVERHAUL_GOLD);
+        case BuildingType::IronMine: return (MiningBehavior)settings.getSelection(AddonId::MINING_OVERHAUL_IRON);
+        case BuildingType::CoalMine: return (MiningBehavior)settings.getSelection(AddonId::MINING_OVERHAUL_COAL);
+        case BuildingType::GraniteMine: return (MiningBehavior)settings.getSelection(AddonId::MINING_OVERHAUL_GRANITE);
+        default: return MiningBehavior::Normal;
+    }
+}
+
+bool nofMiner::StartWorking()
+{
+    isAlteredWorkcycle = false;
+
+    MiningBehavior addonSettings = GetMiningBehavior();
+
+    switch (addonSettings)
+    {
+        case MiningBehavior::S4Like:
+        {
+            int sumResAmount = 0;
+
+            MapPoint nonMinumResPt;
```

Comment:
Did you mean `nonMinimumResPt`? Also no need for the empty line between those 2 variable initializations.